sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -59,7 +59,7 @@ object Cast {
     case (StringType, TimestampType) => true
     case (BooleanType, TimestampType) => true
     case (DateType, TimestampType) => true
-    case (_: NumericType, TimestampType) => true
+    case (_: NumericType, TimestampType) => SQLConf.get.allowCastNumericToTimestamp
```

Comment:
we can avoid adding this method, just do `SQLConf.get.getConf(SQLConf.LEGACY_AllOW_CAST_NUMERIC_TO_TIMESTAMP)`